llvm/lib/CodeGen/TargetLoweringObjectFileImpl.cpp

```diff
@@ -875,9 +875,13 @@ static MCSectionRepo *selectRepoSectionForGlobal(MCContext &Ctx,
     // Add new created digest to the TicketNodes.
     assert(!GO->getMetadata(LLVMContext::MD_repo_ticket) &&
            "TicketNode should be NULL!");
-    auto TN =
-        TicketNode::getIfExists(GO->getParent()->getContext(), GO->getName(),
-                                Result.first, GO->getLinkage(), true);
+    const std::string SymbolName = MCSymbolRepo::getFullName(
+        Ctx, GO->getName(),
+        GlobalValue::isLocalLinkage(GO->getLinkage()) ? Result.first
+                                                      : ticketmd::NullDigest);
+    auto TN = TicketNode::getIfExists(GO->getParent()->getContext(),
```

Comment:
Could we keep `SymbolName` as a `Twine` rather than allocating a `std::string`?

Reply:
We can't. The same reason as [above](https://github.com/SNSystems/llvm-project-prepo/pull/49#discussion_r337116248).